### DISCUSSION TITLE
Concerning power consumption: Is there a way to prohibit Pixi from rendering when nothing changed since latest frame?

### DISCUSSION THREAD

**Discussion by joergplewe** [pixijs/pixijs#9442] [General]
My app has a quite rich content, thousends and thousends of icons and `Graphics`, so - even when idle and nothing changing - consumes 70% of GPU performance and respective electrical power running at 60fps.
It seems (naively, to me) that it makes sense to skip rendering when the scenegraph did not change? Possible?

**Comment by SuperSodaSea** [comment]
You can try setting `autoStart: false` when creating `PIXI.Application`, and call `app.render()` manually when needed. You will need to determine for yourself if any content has changed.

**Reply by joergplewe** [reply]
Thx, but this (again - to me) doesn't sound possible for there are many many places where scenegraph changes take place. 
E.g. a network message coming in or the mouse moving may cause hundreds of scenegraph changes - or none. Additionally `render()` should be called consolidates over potentially many events on time of `rAF`.

**Reply by miltoncandelero** [reply]
It is for that reason that pixi can't detect if/where there has been a scene change and thus always re-renders the entire screen.

It is up to each project to find a way that makes sense on when to halt rendering and when to resume it

Extra pitfall, you need to find a way to detect changes fast enough to not be worse than just rendering anyway :P
In that regard it is similar to the culling problem

**Reply by joergplewe** [reply]
Thx, I suspected it to be like that. :(

I wonder whether something like zone.js could do the trick.

https://github.com/angular/angular/tree/main/packages/zone.js

**Reply by bigtimebuddy** [reply]
Pixi has some optimizations to prevent re-running expensive code (e.g., recomputing transforms) every frame, but it is not a static renderer like you're suggesting. You need to manage renders yourself. There are so many operations a user could make to the scene (e.g., adding filters, masking, manage internal canvases like Text), that Pixi would need to cache a lot more operations. It is just not feasible given the scope of the project and size of the team. Also, traditionally games have been a heavy consumer for this project, and it's rare that something is _not_ moving in these situations.

It helps to know this context from the jump of a project so that you can factor in any visual changes to trigger re-rendering. I created a mostly-static application awhile ago using this approach that bubbles render events and then I debounce render requests at the top-level of my app. Also, you can add render requests to mousemove or other interaction events to the canvas. It's definitely possible, but requires a little more effort.

I hope this helps.

**Reply by joergplewe** [reply]
In fact this is helpful; now I know that I'm not just missing a thing.

Might this change in Pixi8?

**Comment by erikdubbelboer** [comment]
Have you tried looking into [`cacheAsBitmap`](https://pixijs.download/dev/docs/PIXI.DisplayObject.html#cacheAsBitmap) which you can set on most objects such as `Container` and `Graphic`?
